include/zephyr/bluetooth/mesh/access.h

```diff
@@ -670,6 +672,23 @@ struct bt_mesh_mod_id_vnd {
 };
 
 /** Abstraction that describes a Mesh Model instance */
+struct bt_mesh_model_data {
```

Comment:
This structure doesn't participate in any algorithms as a separate logic entity. Why do you need to separate this data? Seems like this adds one more nesting level and more typecasting in general.

Reply:
The biggest reason is that I want to extract the variable variables of this structure, which means that the model is a const when defined.

Reply:
Will be change to this, all models and elements are const, so, will save some memory and safely.

``` C

/* This application only needs one element to contain its models */
static const struct bt_mesh_model models[] = {
	BT_MESH_MODEL_CFG_SRV,
	BT_MESH_MODEL_HEALTH_SRV(&health_srv, &health_pub),
	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op, NULL,
		      NULL),
	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op, NULL,
		      NULL),
};

static const struct bt_mesh_elem elements[] = {
	BT_MESH_ELEM(0, models, BT_MESH_MODEL_NONE),
};

static const struct bt_mesh_comp comp = {
	.cid = BT_COMP_ID_LF,
	.elem = elements,
	.elem_count = ARRAY_SIZE(elements),
};


```